src/scrapeFieldRainoutInfo.ts

```diff
@@ -0,0 +1,167 @@
+import xlsx from "node-xlsx";
+
+const FIELD_RAINOUT_INFO_URL =
+  "https://fs18.formsite.com/res/resultsReportTable?EParam=B6fiTn%2BRcO5gxPxCLTtif%2FTBZfQjAcbzwCi9jw02kUQSq0hsyAedVkhBtIa3wGQcGW07E1SN8yI%3D";
+const FIELD_RAINOUT_EXPORT_URL =
+  "https://fs18.formsite.com/res/resultsReportExport?EParam=B6fiTn-RcO5gxPxCLTtif_TBZfQjAcbzwCi9jw02kUQSq0hsyAedVkhBtIa3wGQcGW07E1SN8yI";
+
+enum FieldComplex {
+  ALL_GRASS_FIELDS_AND_DIAMONDS = "all grass fields & diamonds",
+  OTHER_FIELDS = "other fields",
+  ALL_FIELDS_EXCEPT = "all fields except (see next)",
+  POLO_FIELDS = "polo fields",
+  OTHER_SEE_BELOW = "other see below",
+}
+
+enum ClosedFieldStatus {
+  CLOSED = "closed",
+  ALL_CLOSED = "closed all fields closed",
+}
+
+enum OpenFieldStatus {
+  OPEN = "open",
+  OPEN_FIELDS = "open fields",
+}
+
+const CLOSED_STATUSES = Object.values(ClosedFieldStatus);
+const OPEN_STATUSES = Object.values(OpenFieldStatus);
+
+export const downloadFieldRainoutInfo = async () => {
+  /**
+   * Fetch the field rainout info as an XLSX.
+   *
+   * This request fails if we don't set cookies in the header, so we do a base request first,
```

Comment:
I've used the fetch-cookie package for this in the past and it worked pretty well, bikereg recently added a cookie-gate to their site but since fetch follows redirects transparently it will work given a cookie jar. Although in this case it looks like it returns html with a 200 status on failure so would still require two explicit requests